docs/docs/themes/api-reference.md

```diff
@@ -56,39 +55,86 @@ module.exports = themeOptions => {
+
+### What path do use to shadow a file?
 
-```js:title=src/gatsby-theme-tomato/components/profile-card.js
-import React from "react"
+Until we build tooling to support automatically handling shadowing you will have to figure out how to convert paths in a theme into paths in your site on your own. Luckily the way to do that is few steps. Take the `src` directory from the theme, and move it to the front of the path, then write a file at that location in your site. Looking back on our `Header` example, this is the path to the component in our theme:
 
-export default props => <pre>{JSON.stringify(props, null, 2)}</pre>
 ```
+gatsby-theme-amazing/src/components/header.js
+```
+
+and here is the path where we would shadow it in our site:
+
+```
+<our-site>/src/gatsby-theme-amazing/components/header.js
+```
+
+Shadowing only works on imported files in the `src` directory. This is because shadowing is built on top of Webpack, so the module graph needs to include the shadowable file.
+
+Since we can use many themes in a given site, there are many potential places to shadow a given file (one for each theme and one for the user's site). In the event that there exist many files attempting to shadow `gatsby-theme-amazing/src/components/header.js`, the last one included in the plugins array will win although the user's site will beat any plugin.
 
 ## Theme composition
 
-A theme can declare another theme as a parent theme. This means that a theme can declare another theme in its
-own `gatsby-config.js`. So if you'd like to use `gatsby-theme-blog` to build off of you can and install the theme
-and then configure it:
+Gatsby themes can compose horizontally and vertically. Vertical composition refers to the classic "parent/child" relationship. A child theme declares a parent theme in the child theme's plugins array.
 
-```js:title=gatsby-config.js
+```js:title=gatsby-theme-child/gatsby-config.js
+module.exports = {
+  plugins: [`gatsby-theme-parent`],
+}
+```
+
+Horizontal composition is when two different themes are used together, such as `gatsby-theme-blog` and `gatsby-theme-notes`.
+
+```js:title=my-site/gatsby-config.js
 module.exports = {
-  plugins: ["gatsby-theme-blog"],
+  plugins: [`gatsby-theme-blog`, `gatsby-theme-notes`],
 }
 ```
+
+Themes at their core are an algorithm that merges multiple `gatsby-config.js` files together into a single config your site can use to build with. To do that we need to define how to combine two `gatsby-config.js`s together. Before we can do that, we need to flatten the parent/child relationships into a single array. This results in the final ordering when considering which shadowing file to use if multiple are available.
+
+Our first example results in a final ordering of `['gatsby-theme-parent', 'gatsby-theme-child']` (parents always come before their children so that children can override functionality), while our second example results in `['gatsby-theme-blog', 'gatsby-theme-core']`.
```

Comment:
```suggestion
Our first example results in a final ordering of `['gatsby-theme-parent', 'gatsby-theme-child']` (parents always come before their children so that children can override functionality), while our second example results in `['gatsby-theme-blog', 'gatsby-theme-notes']`.
```

Reply:
whoa, this `core` reference totally came out of my hands without realizing I was writing it.